### PR TITLE
Switching to prototype object for ansi2html and adding render method.

### PR DESCRIPTION
**Pull request by DanBUK**
As title pretty much..

### CODE REVIEW

**Comment by agnoster** [comment]
Just about to head to bed, so I'll look at this tomorrow. Would be nice to know what the motivations/goals for the switch are.

**Comment by DanBUK** [comment]
Main motivations are that I can see this object turning into an pipe or a "render process" object, wanted to turn the usage into an object that persists (in certain long running cases) rather than a function call each time.

**Comment by jprichardson** [comment]
I'm not sure that I see the value here either... ?

File: lib/index.js [hunk 1]
```diff
@@ -1,65 +1,69 @@
 (function(){
-  function declare(module_name, exports) {
-    if (module) module.exports = exports
-    else window[module_name] = exports
+  var declare = function declare (module_name, exports) {
```
Comment (agnoster) [inline lib/index.js:2]: Is this just a stylistic thing? Shouldn't these be basically equivalent?

**Comment by agnoster** [comment]
Finally getting around to reviewing (sorry this fell by the wayside so long!) and I noticed it's really hard to see the actual changes because there's a lot of formatting changes (inserting semicolons, adding var declarations to functions, etc.)

I generally try to follow [npm style](https://npmjs.org/doc/coding-style.html), so I'm really not keen on merging a feature and having it change the style of the project. I'm sure you'd feel similarly if I submitted a pull request removing all the unnecessary semicolons from one of your projects ;-)

I like the idea though. If you don't feel like undoing the formatting changes, I'd be happy to adapt it at some point (though it might take some time).

Just in general for future reference, I tend to think it's best to stick with a project's formatting, and to use that in any new code I submit to it (and certainly not to gratuitously change style in existing code). Maybe it was an innocent mistake caused by an overzealous IDE or something ;-)

**Pull request closed without merging**